### PR TITLE
Balder/add support for trace in accesslog

### PR DESCRIPTION
**Pull request by baldersheim**
@bjorncs and @bratseth PR

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java [hunk 1]
```diff
@@ -410,7 +413,7 @@ private Result search(String request, Query query, Chain<Searcher> searchChain)
         } catch (ParseException e) {
             ErrorMessage error = ErrorMessage.createIllegalQuery("Could not parse query [" + request + "]: "
                                                                  + Exceptions.toMessageString(e));
-            log.log(LogLevel.DEBUG, () -> error.getDetailedMessage());
```
Comment (bratseth) [inline container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java]: Why are you removing the function indirection in these?

File: container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java [hunk 2]
```diff
@@ -788,11 +642,11 @@ protected void renderFieldContents(Object field) throws IOException {
             if (field instanceof Inspectable && ! (field instanceof FeatureData)) {
                 renderInspector(((Inspectable)field).inspect());
             } else {
-                renderFieldContentsDirect(field);
+                consume(field);
             }
         }
 
-        private void renderFieldContentsDirect(Object field) throws IOException {
+        public void consume(Object field) throws IOException {
```
Comment (bratseth) [inline container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java]: Isn't the naming convention here "acceptX"?

**Review by bratseth — COMMENTED** [review]
Otherwise LGTM

File: container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java (hunk 1 above)
Comment (baldersheim) [reply container-search/src/main/java/com/yahoo/search/handler/SearchHandler.java]: That was supposed to be method reference, not method call. Fixed

File: container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java (hunk 2 above)
Comment (baldersheim) [reply container-search/src/main/java/com/yahoo/search/rendering/JsonRenderer.java]: Correct -> fixed

File: container-accesslogging/src/main/java/com/yahoo/container/logging/TraceRenderer.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
+package com.yahoo.container.logging;
```
Comment (bjorncs) [inline container-accesslogging/src/main/java/com/yahoo/container/logging/TraceRenderer.java]: Missing copyright header and author.

File: container-accesslogging/src/main/java/com/yahoo/container/logging/TraceRenderer.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+import com.yahoo.yolean.trace.TraceNode;
+import com.yahoo.yolean.trace.TraceVisitor;
+import com.fasterxml.jackson.core.JsonGenerator;
+
+import java.io.IOException;
+
+public class TraceRenderer extends TraceVisitor {
```
Comment (bjorncs) [inline container-accesslogging/src/main/java/com/yahoo/container/logging/TraceRenderer.java]: Consider `TraceRenderer` to yolean (unless it has dependencies to types in `container-accesslogging`). Both `TraceVisitor` and `TraceNode` are classes from yolean.

File: container-accesslogging/src/main/java/com/yahoo/container/logging/TraceRenderer.java [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    }
+
+    boolean isInsideOpenObject() {
+        return insideOpenObject;
+    }
+
+    void setInsideOpenObject(boolean insideOpenObject) {
+        this.insideOpenObject = insideOpenObject;
+    }
+    public static final class TraceRenderWrapper extends RuntimeException {
```
Comment (bjorncs) [inline container-accesslogging/src/main/java/com/yahoo/container/logging/TraceRenderer.java]: This could be replaced with `java.io.UncheckedIOException`

**Review by bjorncs — COMMENTED** [review]
LGTM, see comments

**Pull request merged**